echo/wavefront-tracing-utils.go

```diff
@@ -0,0 +1,173 @@
+	var durationSampler tracer.DurationSampler
+	var rateSampler tracer.RateSampler
+
+	//Initialising tracer with RateSampler and DurationSampler if given in conifg
+	//If values for RateSampler and DurationSampler is not given, it initalises standard tracer with default values
+	if tracerConfig.DurationSampler > 0 && tracerConfig.RateSampler > 0 {
+		durationSampler = tracer.DurationSampler{Duration: time.Duration(tracerConfig.DurationSampler) * time.Second}
+		rateSampler = tracer.RateSampler{Rate: tracerConfig.RateSampler}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(rateSampler), wfTracer.WithSampler(durationSampler))
+	} else if tracerConfig.RateSampler > 0 {
+		rateSampler = tracer.RateSampler{Rate: tracerConfig.RateSampler}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(rateSampler))
+	} else if tracerConfig.DurationSampler > 0 {
+		durationSampler = tracer.DurationSampler{Duration: time.Duration(tracerConfig.DurationSampler) * time.Second}
+
+		Tracer = wfTracer.New(reporter, wfTracer.WithSampler(durationSampler))
+	} else {
+		Tracer = wfTracer.New(reporter)
+	}
+
+	//Instantiating Global Tracer
+	opentracing.InitGlobalTracer(Tracer)
+
+	//Enabling Middleware
+	echoWeb.Use(middlewareTracing)
+	log.Println("Tracer Initialized...")
+	return nil
+}
+
+//StartTraceSpan looks for existing context from the headers of the request.
+//If context is found, it starts a child span to the parent span,
+//Else it starts a root span.
+//Tags are added to the respective spans created.
+//If parent span exists parent span id is returned along with child span otherwise root span and "" is returned
+func StartTraceSpan(c echo.Context, appSpecificOperationName string, tags map[string]string) (opentracing.Span, string, error) {
+	var serverSpan opentracing.Span
+	var parentSpanID string
+	wireContext, err := opentracing.GlobalTracer().Extract(
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(c.Request().Header))
+	if err != nil {
+		log.Println(err)
+	}
+
+	//Adding parent span Id if exists
+	sc, ok := wireContext.(tracer.SpanContext)
+	if ok {
+		parentSpanID = sc.SpanID
+	}
+
+	// Create the span referring to the RPC client if available.
+	// If wireContext == nil, a root span will be created.
+	serverSpan = opentracing.StartSpan(
+		appSpecificOperationName,
+		ext.RPCServerOption(wireContext))
+
+	//Populating Tags
+	for key, value := range tags {
+		if value != "" {
+			serverSpan.SetTag(key, value)
+		}
+	}
+
+	return serverSpan, parentSpanID, nil
+}
+
+//InjectTracerHTTP injects the current span into the tracer.
+//Adds the context of the current span to the headers of the Http request
+//to the previous span of the tracer.
+func InjectTracerHTTP(tracer opentracing.Tracer, serverSpan opentracing.Span, httpReq *http.Request) *http.Request {
+	tracer.Inject(
+		serverSpan.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(httpReq.Header))
+	return httpReq
```

Comment:
Why return the request if the user already has a handle to it? Just make this a void func?